library/src-bootstrapped/dotty/internal/StringContextMacro.scala

```diff
@@ -63,12 +63,12 @@ object StringContextMacro {
 
     val (partsExpr, parts) = strCtxExpr match {
       case Expr.StringContext(p1 @ Consts(p2)) => (p1.toList, p2.toList)
-      case _ => qctx.throwError("Expected statically known String Context", strCtxExpr)
+      case _ => Reporting.throwError("Expected statically known String Context", strCtxExpr)
     }
 
     val args = argsExpr match {
       case Varargs(args) => args
-      case _ => qctx.throwError("Expected statically known argument list", argsExpr)
+      case _ => Reporting.throwError("Expected statically known argument list", argsExpr)
```

Comment:
It seems to me the usability is better if methods can be called directly on `qctx`, as they don't need to remember another top-level name. What's the consideration to introduce `Reporting`?

Reply:
It can be imported if needed. The previous location was unintuitive and a few missed it. Additionally, this seems to be the only reason to not use anonymous `QuoteContext`s.

Reply:
What about using the same trick as Dotty to introduce a method `def qctx(using QuoteContext) = ...`?

Reply:
That would also help. But, still errors and warnings do not belong there.

Reply:
I agree with you, logically it is better to separate things.

The little worry I have is that Scala 3 macro authors may also be compiler hackers, compiler plugin writers, and some experience with Scala 2 macros. In all other cases, the error reporting is the same by calling `ctx.error`. So it would be nice to follow the convention.

Reply:
As shown, that can be optionally supported with an implicit conversion.

It also seems that the users of the API did not find the reporting utilities or ended up finding the ones in the reflection API. The second is fine but the overhead could have been avoided by making it more explicit.